### ISSUE TITLE
abortOnError enhancement request

### ISSUE THREAD

**Issue by mreuvers** [ezolenko/rollup-plugin-typescript2#168]
Hello there,

Currently the option abortOnError: true aborts on the first error it encounters when checking a typescript file. When set to false it doesn't bail out at all (and often succeeds completely, even though typescript reported several semantic errors in multiple files e.g.).

I'd like an in-between option, that checks all typescript files, gathers the errors reported, and if there is any error at all (or more) spits it all out at once at the end - and aborts.

Currently the abortOnError: true is painful to work with with (many) errors in large projects, and not aborting at all, is just wrong here as the resulting rollup bundle will be broken.

Would something like this be possible? Thank you.

**Comment by ezolenko** [comment]
Yep. Might get messy due to the way rollup detects failure (see continueAfterFirstError branch if you want to test it out).

**Comment by mreuvers** [comment]
Thank you! I'll give it a try hopefully later today and let you know how it works for me. 

Out of curiousity: I saw you also added an additional flag abortOnWarning, what kind of warnings would it abort on in this case?

**Comment by mreuvers** [comment]
It seems to work first test. I'd like to test a bit more coming days if you're not in a rush with it?
